python/felis/metadata.py

```diff
@@ -0,0 +1,500 @@
+
+        # Create the SQA table object and its columns.
+        name = table_obj.name
+        id = table_obj.id
+        description = table_obj.description
+        columns = [self.build_column(column) for column in table_obj.columns]
+        table = Table(
+            name,
+            self.metadata,
+            *columns,
+            comment=description,
+            schema=self.schema.name if self.apply_schema_to_tables else None,
+            **optargs,  # type: ignore[arg-type]
+        )
+
+        # Create the indexes and add them to the table.
+        indexes = [self.build_index(index) for index in table_obj.indexes]
+        for index in indexes:
+            index._set_parent(table)
+            table.indexes.add(index)
+
+        self._objects[id] = table
+
+    def build_column(self, column_obj: dm.Column) -> Column:
+        """Build a SQA column from a `felis.datamodel.Column` object.
+
+        Parameters
+        ----------
+        column_obj : `felis.datamodel.Column`
+            The column object from which to build the SQA column.
+
+        Returns
+        -------
+        column: `Column`
+            The SQA column object.
+        """
+        # Get basic column attributes.
+        name = column_obj.name
+        id = column_obj.id
+        description = column_obj.description
+        default = column_obj.value
+
+        # Handle variant overrides for the column (e.g., "mysql:datatype").
+        datatype = get_datatype_with_variants(column_obj)
+
+        # Set default value of nullable based on column type and then whether
+        # it was explicitly provided in the schema data.
+        nullable_default = False if isinstance(datatype, Numeric) else True
+        nullable = column_obj.nullable if column_obj.nullable is not None else nullable_default
+
+        # Set autoincrement depending on if it was provided explicitly.
+        autoincrement: Literal["auto", "ignore_fk"] | bool = (
+            column_obj.autoincrement if column_obj.autoincrement else "auto"
+        )
+
+        column: Column = Column(
+            name,
+            datatype,
+            comment=description,
+            autoincrement=autoincrement,
+            nullable=nullable,
+            server_default=default,
+        )
+
+        self._objects[id] = column
+
+        return column
+
+    def build_constraints(self) -> None:
+        """Build the SQA constraints in the Felis schema and append them to the
+        associated `Table`.
+        """
+        for table_obj in self.schema.tables:
+            table = self._objects[table_obj.id]
+            for constraint_obj in table_obj.constraints:
+                constraint = self.build_constraint(constraint_obj)
+                table.append_constraint(constraint)
+
+    def build_constraint(self, constraint_obj: dm.Constraint) -> Constraint:
+        """Build a SQA `Constraint` from a `felis.datamodel.Constraint` object.
```

Comment:
> I think you need to build the docs and check for bad links. Consider running with `package-docs -n build` and it might tell you all the problems.

I'll find and fix these types of errors in the context of another ticket, as the Felis site doesn't even include the Sphinx API documentation right now. So that configuration needs to be done first before `package-docs` even includes it. 

I did fix all the bad typing references you mentioned, as well as a few others that I found.